tuf/scripts/repo.py

```diff
@@ -282,24 +285,35 @@ def delegate(parsed_arguments):
       os.path.join(parsed_arguments.path, REPO_DIR))
 
   if parsed_arguments.role == 'targets':
-    repository.targets.delegate(parsed_arguments.delegatee, public_keys,
-        parsed_arguments.delegate, parsed_arguments.threshold,
-        parsed_arguments.terminating, list_of_targets=None,
-        path_hash_prefixes=None)
+    parent_targets = repository.targets
+    # Use key from --sign if present or use the default targets_key
+    parent_targets_keys = parsed_arguments.sign or [os.path.join(KEYSTORE_DIR, TARGETS_KEY_NAME)]
+  else:
+    parent_targets = repository.targets(parsed_arguments.role)
+    parent_targets_keys = parsed_arguments.sign
 
+  parent_targets.delegate(parsed_arguments.delegatee, public_keys,
+    parsed_arguments.delegate, parsed_arguments.threshold,
+    parsed_arguments.terminating, list_of_targets=None,
+    path_hash_prefixes=None
+  )
+
+  # Load the private key for the parent targets
+  for key in parent_targets_keys:
     targets_private = import_privatekey_from_file(
-        os.path.join(parsed_arguments.path, KEYSTORE_DIR, TARGETS_KEY_NAME),
-        parsed_arguments.targets_pw)
+      os.path.join(parsed_arguments.path, key),
+      parsed_arguments.targets_pw
+    )
+    parent_targets.load_signing_key(targets_private)
 
-    repository.targets.load_signing_key(targets_private)
+  # Load the private keys of the delegatee targets to allow the generation of the delegatee targets file
+  for key in parsed_arguments.pubkeys:
+    delegatee_private = import_privatekey_from_file(
+      os.path.join(parsed_arguments.path, key[:-4]),
```

Comment:
This now loads private keys based on `--pubkeys` argument, making assumptions on filenames that are not made before this in the code. Hmm...

I have to admit I don't care too much about the maintainability of this file (that train has probably passed the station)... I'm just pointing out that that looks like a workaround.

Reply:
Yes completely. Otherwise I would have had to update the interface

But yeah overall I feel like to have something clean a lot of things should probably be rewritten ^^'

Reply:
Actually an assumption is already done in the code that the key should be named base on the role (targets.pub/snapshot.pub) otherwise it doesn't work. Which actually make things complicated if you have multiple key and a threshold

Reply:
It's not the worst assumption as long as we clearly document it. Can we set the expected filename in a well-named variable, along with a comment?

Reply:
yes I put a comment!